### PR TITLE
Add unit test utilities temporary_locale and TestCaseWithCLocale

### PR DESCRIPTION
**Pull request by boxydog**
# Pull Request Checklist

Resolves: #1347 
Resolves: #2964 
Resolves: #2815  

- [X] Ensured **tests pass** and (if applicable) updated functional test output
- [X] Conformed to **code style guidelines** by running appropriate linting tools
- [X] Added **tests** for changed code
- [X] Updated **documentation** for changed code

Note:

This fails the new pre-commit check for ruff-format, but I didn't want this PR clouded by a huge reformat. I would push that to a different PR.

The failure:

```
ruff-format..............................................................Failed
- hook id: ruff-format
- exit code: 1

warning: `ruff format` is not yet stable, and subject to change in future versions.
Would reformat: pelican/utils.py
Would reformat: pelican/tests/test_utils.py
Would reformat: pelican/tests/test_importer.py
Would reformat: pelican/tests/test_generators.py
4 files would be reformatted
```

This is like #2964, but with conflicts resolved, and more locale statements factored out.

### CODE REVIEW

File: pelican/utils.py [hunk 1]
```diff
@@ -999,3 +996,31 @@ def maybe_pluralize(count, singular, plural):
...
+    Use TestCaseWithCLocale if you want every test to use the C locale.
+    '''
+    orig_locale = locale.setlocale(lc_category)
+    if temp_locale:
+        locale.setlocale(lc_category, temp_locale)
+    yield orig_locale
+    locale.setlocale(lc_category, orig_locale)
+
+
+class TestCaseWithCLocale(unittest.TestCase):
```
Comment (avaris) [inline pelican/utils.py:1016]: This is a unittest helper rather than a general pelican util and thus should go into `pelican.tests.support`.

**Review by avaris — CHANGES_REQUESTED** [review]

File: pelican/utils.py (hunk 1 above)
Comment (boxydog) [reply pelican/utils.py:1016]: Moved.

**Review by avaris — APPROVED** [review]
Thanks!

**Review by justinmayer — APPROVED** [review]
Many thanks to @boxydog for this excellent enhancement and to @avaris for reviewing. 💯

**Pull request merged**